pkg/engine2/operational_eval/vertex_resource_rule_test.go

```diff
@@ -0,0 +1,220 @@
+package operational_eval
+
+import (
+	"fmt"
+	"testing"
+
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+	"github.com/stretchr/testify/assert"
+	gomock "go.uber.org/mock/gomock"
+)
+
+func Test_resourceRuleVertex_Key(t *testing.T) {
+	tests := []struct {
+		name string
+		v    resourceRuleVertex
+		want Key
+	}{
+		{
+			name: "resource rule vertex",
+			v: resourceRuleVertex{
+				Resource: construct.ResourceId{Name: "test"},
+				Rule: knowledgebase.AdditionalRule{
+					If: "test",
+				},
+			},
+			want: Key{
+				Ref: construct.PropertyRef{
+					Resource: construct.ResourceId{Name: "test"},
+				},
+				RuleHash: "9a510ea0226e156085c51625d22ac3679eb7530fd36ba925fcbe211da9c1b373",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			got := tt.v.Key()
+			assert.Equal(tt.want, got)
+		})
+	}
+}
+
+func Test_resourceRuleVertex_Dependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	dcap := NewMockdependencyCapturer(ctrl)
+
+	tests := []struct {
+		name     string
+		resource construct.ResourceId
+		rule     knowledgebase.AdditionalRule
+		mocks    func()
+		wantErr  bool
+	}{
+		{
+			name:     "resource rule vertex",
+			resource: construct.ResourceId{Name: "test"},
+			rule: knowledgebase.AdditionalRule{
+				If: "test",
+			},
+			mocks: func() {
+				dcap.EXPECT().ExecuteOpRule(knowledgebase.DynamicValueData{
+					Resource: construct.ResourceId{Name: "test"},
+				}, knowledgebase.OperationalRule{
+					If: "test",
+				}).Return(nil)
+			},
+		},
```

Comment:
Missing a test case with other fields in the `AdditionalRule` to make sure those get translated into `OperationalRule` properly.